packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -68,60 +67,59 @@ const GridVisualizerGrid = forwardRef(
 			};
 		}, [ gridElement ] );
 
-		useEffect( () => {
-			function onGlobalDrag() {
-				setIsDroppingAllowed( true );
-			}
-			function onGlobalDragEnd() {
-				setIsDroppingAllowed( false );
-			}
-			document.addEventListener( 'drag', onGlobalDrag );
-			document.addEventListener( 'dragend', onGlobalDragEnd );
-			return () => {
-				document.removeEventListener( 'drag', onGlobalDrag );
-				document.removeEventListener( 'dragend', onGlobalDragEnd );
-			};
-		}, [] );
-
 		return (
-			<BlockPopoverCover
-				className={ clsx( 'block-editor-grid-visualizer', {
-					'is-dropping-allowed': isDroppingAllowed,
-				} ) }
-				clientId={ gridClientId }
-				__unstablePopoverSlot="__unstable-block-tools-after"
-			>
-				<div
-					ref={ ref }
-					className="block-editor-grid-visualizer__grid"
-					style={ gridInfo.style }
-				>
-					{ isManualGrid ? (
-						<ManualGridVisualizer
-							gridClientId={ gridClientId }
-							gridInfo={ gridInfo }
-						/>
-					) : (
-						Array.from( { length: gridInfo.numItems }, ( _, i ) => (
-							<GridVisualizerCell
-								key={ i }
-								color={ gridInfo.currentColor }
-							/>
-						) )
-					) }
-				</div>
-			</BlockPopoverCover>
+			<>
+				<GridPopunder
+					gridClientId={ gridClientId }
+					gridInfo={ gridInfo }
+					isManualGrid={ isManualGrid }
+				/>
+				{ isManualGrid && (
+					<GridPopover
+						ref={ ref }
```

Comment:
There's a bug here caused by `ref` only being passed/set when `isManualGrid` is true. It means that you're able to resize a grid item beyond the bounds of the grid block when it's in auto mode.

https://github.com/user-attachments/assets/ede25412-04c1-47a4-a3d9-de2c2fa3df37

Reply:
Should be fixed in f73ca9f. The annoying aspect is that it means some parts of `GridPopover` always have to be rendered even when there's not much going on, but I think it's necessary. I tried to optimize it by moving the items into a separate component.

To properly finish https://github.com/WordPress/gutenberg/issues/64162, the auto grid will also need a single appender in the next available cell, so something like this will be needed anyway.

Reply:
> To properly finish https://github.com/WordPress/gutenberg/issues/64162, the auto grid will also need a single appender in the next available cell, so something like this will be needed anyway.

I think we can ignore that tbh. It's better imo to have it consistent across container blocks.